src/components/SignUpForm/index.jsx

```diff
@@ -1,5 +1,150 @@
+    "string.empty": `Please provide your first name`,
+  }),
+  lastName: Joi.string().required().messages({
+    "string.empty": `Please provide your last name`,
+  }),
+  email: Joi.string()
+    .required()
+    .email({ tlds: { allow: ["com", "net", "edu"] } })
+    .messages({
+      "string.empty": `Please provide your email`,
+      "string.email": `Please provide a valid email`,
+    }),
+  password: Joi.string().required().min(8).max(64).strict().messages({
+    "string.empty": `Please provide a password`,
+  }),
+  confirmPassword: Joi.string()
+    .valid(Joi.ref("password"))
+    .required()
+    .strict()
+    .messages({
+      "any.only": "Those passwords didn't match",
+    }),
+  acceptTerms: Joi.valid("accepted").required().messages({
+    "any.only": `You should accept our terms of service and privcey policy`,
+  }),
+});
+
+export default function SignUpForm({ submit }) {
+  const { register, handleSubmit, errors, reset } = useForm({
+    resolver: joiResolver(schema),
+  });
+
+  const onSubmit = (data) => {
+    submit(data);
+    reset();
+  };
+
+  return (
+    <div className="signUpForm">
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <div className="formInputGroup">
+          <div className="formInput">
+            <input
+              className={errors.email && "inputError"}
+              type="text"
+              placeholder="First name"
+              name="firstName"
+              aria-label="firstName"
+              ref={register()}
+            />
+            {errors?.firstName && (
+              <InputErrorMessage message={errors.firstName.message} />
+            )}
+          </div>
+
+          <div className="formInput">
+            <input
+              className={errors.email && "inputError"}
+              type="text"
+              placeholder="Last name"
+              name="lastName"
+              aria-label="lastName"
+              ref={register()}
+            />
+            {errors?.lastName && (
+              <InputErrorMessage message={errors.lastName.message} />
+            )}
+          </div>
+        </div>
+
+        <div className="formInput">
+          <input
+            className={errors.email && "inputError"}
+            type="email"
+            placeholder="Your email"
+            name="email"
+            aria-label="email"
+            ref={register()}
+          />
+          {errors?.email && (
+            <InputErrorMessage message={errors.email.message} />
+          )}
+        </div>
+
+        <div className="formInputGroup">
+          <div className="formInput">
+            <input
+              className={errors.password && "inputError"}
+              type="password"
+              placeholder="Password"
+              name="password"
+              aria-label="password"
+              ref={register()}
+            />
+            {errors?.password && (
+              <InputErrorMessage message={errors.password.message} />
```

Comment:
all these can be folded into a common fn or component that takes errors?.x and if it's non null it renders errors?.x.messages. then reused throughout all your components.

try to find where you are copy pasting this pattern and fix

Reply:
done!